File: frontend/src/lib/components/definitions/tripletap.ts
import { ComponentDefinition } from '../types'
import { CommandBuilders } from '../utils'

export const tripletapComponent: ComponentDefinition = {
  id: 'tripletap',
  name: 'Tripletap',
  description: 'Start xovi by triple-pressing the power button',
  version: 'main',
  author: 'rmitchellscott',

  dependencies: ['xovi'],

  checkInstalled: () =>
    CommandBuilders.testExists('/home/root/xovi-tripletap/uninstall.sh'),

  install: () => ({
    script:
      'cd /home/root && wget -qO- https://raw.githubusercontent.com/rmitchellscott/xovi-tripletap/main/install.sh | bash',
    description: 'Download and run tripletap installer',
  }),

  uninstall: () => ({
    script: '/home/root/xovi-tripletap/uninstall.sh',
    description: 'Run tripletap uninstaller',
  }),

  maintenanceCommands: [
    {
      id: 'enable',
      label: 'Re-enable After Updates',
      description: 'Re-enable tripletap service (run after software updates)',
      command: () => ({
        script: '/home/root/xovi-tripletap/enable.sh',
        description: 'Enable tripletap service',
      }),
    },
    {
      id: 'disable',
      label: 'Disable Service',
      description: 'Stop and disable the tripletap systemd service',
      command: () => ({
        script: 'systemctl disable xovi-tripletap --now',
        description: 'Disable tripletap service',
      }),
    },
  ],

  category: 'enhancement',
  tags: ['activation', 'convenience'],
  requiresReboot: false,
}

File: internal/component/definitions/tripletap.go
package definitions

import (
	"reManager/internal/commands"
	"reManager/internal/component"
)

var TripletapComponent = &component.ComponentDefinition{
	ID:          "tripletap",
	Name:        "Tripletap",
	Description: "Start xovi by triple-pressing the power button",
	Version:     "main",
	Author:      "rmitchellscott",

	Dependencies: []string{"xovi"},

	CheckInstalled: func(ctx component.CommandContext) component.CommandResult {
		return commands.TestExists("/home/root/xovi-tripletap/uninstall.sh", commands.PathFile)
	},

	Install: func(ctx component.CommandContext) []component.CommandResult {
		return []component.CommandResult{
			{
				Script:      "cd /home/root && wget -qO- https://raw.githubusercontent.com/rmitchellscott/xovi-tripletap/main/install.sh | bash",
				Description: "Download and run tripletap installer",
			},
		}
	},

	Uninstall: func(ctx component.CommandContext) []component.CommandResult {
		return []component.CommandResult{
			{
				Script:      "/home/root/xovi-tripletap/uninstall.sh",
				Description: "Run tripletap uninstaller",
			},
		}
	},

	MaintenanceCommands: []component.MaintenanceCommand{
		{
			ID:          "enable",
			Label:       "Re-enable After Updates",
			Description: "Re-enable tripletap service (run after software updates)",
			Command: func(ctx component.CommandContext) []component.CommandResult {
				return []component.CommandResult{
					{
						Script:      "/home/root/xovi-tripletap/enable.sh",
						Description: "Enable tripletap service",
					},
				}
			},
		},
		{
			ID:          "disable",
			Label:       "Disable Service",
			Description: "Stop and disable the tripletap systemd service",
			Command: func(ctx component.CommandContext) []component.CommandResult {
				return []component.CommandResult{
					{
						Script:      "systemctl disable xovi-tripletap --now",
						Description: "Disable tripletap service",
					},
				}
			},
		},
	},

	Category:       "enhancement",
	Tags:           []string{"activation", "convenience"},
	RequiresReboot: false,
}
